File: sudoku.py
class Solution(object):
    def isValidSudoku(self, board):
        """
        :type board: List[List[str]]
        :rtype: bool
        """
        unique = set()

        for i in range(9):
            for j in range(9):
                if board[i][j]!='.':
                    b = "("+board[i][j]+")"
                    row = str(i)+b
                    col = b+str(j)
                    box = str(i/3) + b + str(j/3)
                    if row in unique or col in unique or box in unique:
                        return False
                    unique.add(row)
                    unique.add(col)
                    unique.add(box)
        return True

File: Arrays and Hashing/sudoku.java
class Solution {
    public boolean isValidSudoku(char[][] board) {
        HashSet<String> set = new HashSet<>();
        for(int i = 0; i<board.length; i++){
            for(int j = 0; j<board[0].length; j++){
                if(board[i][j]!='.'){
                    String b = "("+board[i][j]+")";
                    if(!set.add(i+b) || !set.add(b+j) || !set.add(i/3 + b + j/3)){
                        return false;
                    }
                }
            }
        }
        return true;
    }
}